boards/arm/mps2_an521/board.cmake

```diff
@@ -21,5 +25,5 @@ if (CONFIG_BUILD_WITH_TFM)
 elseif (CONFIG_SOC_MPS2_AN521_CPU1)
   set(CPU0_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/zephyr/boards/arm/mps2_an521/empty_cpu0-prefix/src/empty_cpu0-build/zephyr)
   set(QEMU_KERNEL_OPTION "-device;loader,file=${CPU0_BINARY_DIR}/zephyr.elf")
-  set(QEMU_EXTRA_FLAGS "-device;loader,file=${CMAKE_CURRENT_BINARY_DIR}/zephyr/zephyr.elf")
+  list(APPEND QEMU_EXTRA_FLAGS "-device;loader,file=${CMAKE_CURRENT_BINARY_DIR}/zephyr/zephyr.elf")
```

Comment:
Thanks for the tip!

The problem is that for some reason `set` conflicts with `list`. The runtime error happen because `-device;loader,file=X` was not passed to qemu. With the new patch version I hope everything should work.

Reply:
> The problem is that for some reason set conflicts with list.

A `set` can be used to create a list https://cmake.org/cmake/help/latest/command/set.html
> Multiple arguments will be joined as a [semicolon-separated list](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#cmake-language-lists) to form the actual variable value to be set.

but it will also clear any existing values in the list, for example:
```
set(MY_LIST a b c)
list(APPEND MY_LIST d)
``` 
create the CMake list `a;b;c;d`, whereas:
```
list(APPEND MY_LIST a b c)
set(MY_LIST d)
```
create the CMake list (variable) of only `d`.

but in your former case, the `list(APPEND ...)` came after `set()`, so everything should be ok.
(unless some other code somewhere messes with the `QEMU_EXTRA_FLAGS` variable)

Reply:
I noted that `list(APPEND QEMU_EXTRA_FLAGS "-device;loader,file=${PROJECT_BINARY_DIR}/${KERNEL_ELF_NAME}")` was't added, not sure why, when using `set` for those cases that we saw kernel panic. My feeling is that depending on the combination of quotes and set/list the behavior changes.